content/en/blog/_posts/2023-09-15-Bootstrap-an-Air-Gapped-Cluster-with-Kubeadm/index.md

```diff
@@ -0,0 +1,665 @@
+
+# Prepare the Node for K8s
+
+Before downloading and moving the artifacts to the VM, let's first prep that VM to run Kubernetes.
+
+### As a normal user:
+
+**Make destination directory for software artifacts**
+
+```bash
+mkdir ~/tmp
+```
+### As root:
+
+**Write to /etc/sysctl.d/99-k8s-cri.conf**
+
+```bash
+cat > /etc/sysctl.d/99-k8s-cri.conf << EOF
+net.bridge.bridge-nf-call-iptables=1
+net.ipv4.ip_forward=1
+net.bridge.bridge-nf-call-ip6tables=1
+EOF
+```
+
+**Write to /etc/modules-load.d/k8s.conf**
+```bash
+echo -e overlay\\nbr_netfilter > /etc/modules-load.d/k8s.conf
+```
+**Install iptables-legacy**
+```bash
+dnf -y install iptables-legacy
+```
+**Set iptables to use iptables-legacy**
+```bash
+update-alternatives --set iptables /usr/sbin/iptables-legacy
+```
+**Turn off swap**
+```bash
+touch /etc/systemd/zram-generator.conf
+systemctl mask systemd-zram-setup@.service
+sed -i '/ swap / s/^\(.*\)$/#\1/g' /etc/fstab
+```
+
+**Disable firewalld**
+```bash
+systemctl disable --now firewalld
+```
+**Disable systemd-resolved**
+```bash
+systemctl disable --now systemd-resolved
+```
+**Configure DNS defaults for NetworkManager**
+```bash
+sed -i '/\[main\]/a dns=default' /etc/NetworkManager/NetworkManager.conf
+```
+**Blank the system-level DNS resolver configuration**
+
+```bash
+unlink /etc/resolv.conf || true
+touch /etc/resolv.conf
+```
+
+**Disable SELinux (just for a demo - check before doing this in production!)**
+
+```bash
+setenforce 0
+```
+
+**Make sure all changes survive a reboot**
+```bash
+reboot
+```
+# Download all the artifacts
+
+On the laptop/host machine, download all of the artifacts enumerated in the previous section.  Since the air gapped VM is running Fedora 37, all of the dependencies shown in this part are for Fedora 37.  Note, this procedure will only work on AArch64 or AMD64 CPU architectures.  Execute this procedure anywhere where you have write permission, your home directory is perfectly suitable.
+
+Note, operating system packages for the Kubernetes artifacts that need to be carried across can now be found at [pkgs.k8s.io](https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/). This blog post will use a combination of Fedora repositories and GitHub in order to download all of the required artifacts. When you’re doing this on your own cluster, you should pick whether to use the official Kubernetes packages, or the official packages from your operating system distribution - both are valid choices.
```

Comment:
```suggestion
Note, operating system packages for the Kubernetes artifacts that need to be carried across
can now be found at [pkgs.k8s.io](/blog/2023/08/15/pkgs-k8s-io-introduction/).
This blog post will use a combination of Fedora repositories and GitHub in order to download
all of the required artifacts. When you’re doing this on your own cluster, you should pick
whether to use the official Kubernetes packages, or the official packages from your operating
system distribution - both are valid choices.
```

Reply:
Instead of `pick` I recommend using `decide` as this makes more user-friendly and sensible.

Reply:
For _docs_ I'd agree @Vikash-8090-Yadav. However this is a blog article and it's OK for @TheFutonEng to write in his authentic own choice of words.

@TheFutonEng: your call, on this one.

Reply:
Went with 'decide,' thanks for the suggestions @Vikash-8090-Yadav .